File: 0129-sum-root-to-leaf-numbers/0129-sum-root-to-leaf-numbers.js
/**
 * Definition for a binary tree node.
 * function TreeNode(val, left, right) {
 *     this.val = (val===undefined ? 0 : val)
 *     this.left = (left===undefined ? null : left)
 *     this.right = (right===undefined ? null : right)
 * }
 */
/**
 * @param {TreeNode} root
 * @return {number}
 */
var sumNumbers = function(root) {
    if (!root) return 0;
    
    const res = []

    function traverse(root, num='') {
        if (!root) return
        num+=root.val;
        if (!root.left && !root.right) {
            res.push(num);
            return;
        };
        // console.log(num)
        traverse(root.left,num);
        traverse(root.right,num);

        return

    }

    traverse(root)

    return res.reduce((accum, val) => accum + parseInt(val), 0)
};

File: 0129-sum-root-to-leaf-numbers/0129-sum-root-to-leaf-numbers.py
# Definition for a binary tree node.
# class TreeNode:
#     def __init__(self, val=0, left=None, right=None):
#         self.val = val
#         self.left = left
#         self.right = right
class Solution:
    def sumNumbers(self, root: Optional[TreeNode]) -> int:
        def traverse(root, accum=0):
            if not root:
                return 0

            accum = accum * 10 + root.val
        
            if not root.left and not root.right:
                return accum
            
            return traverse(root.left, accum) + traverse(root.right, accum)

        return traverse(root)